src/go/rpk/pkg/cli/topic/mount.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package topic
+
+import (
+	"time"
+
+	"github.com/redpanda-data/common-go/rpadmin"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newMountCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var alias string
+	var aliasNamespace string
+	var namespace string
+	var location string
+	var timeout int
+
+	cmd := &cobra.Command{
+		Use:   "mount [TOPICS...]",
+		Short: "Mount a topic to the Redpanda Cluster from tiered storage",
+		Long:  helpMount,
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, topics []string) {
+			pf, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+			config.CheckExitCloudAdmin(pf)
+			adm, err := adminapi.NewClient(fs, pf)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			topic := rpadmin.InboundTopic{
+				SourceTopic: rpadmin.Topic{
+					Topic:     topics[0],
+					Namespace: namespace,
+				},
+			}
+			if alias != "" && aliasNamespace != "" {
+				topic.Alias = &rpadmin.Topic{
+					Topic:     alias,
+					Namespace: aliasNamespace,
+				}
+			}
+			if location != "" {
+				topic.Location = location
+			}
+
+			migration, err := adm.AddInboundMigration(cmd.Context(), rpadmin.InboundMigration{
+				MigrationType:  "inbound",
```

Comment:
It's a fair point. My initial intent in the common-go PR was to have a single function that accepted any, type checked the input to ensure it was valid and reused it to pass along to the API. With the requested change to specific functions in that PR I opted to continue using the outbound and inbound structs to avoid duplication of the structs to eliminate a single field. 

I can go back to make a further change to eliminate said field, but I do feel that it'd be adding quite a bit of code to avoid using any.